File: web/api/schemas/pipeline.py
"""Pipeline response schemas."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class PipelineNode(BaseModel):
    id: str = ""
    title: str = ""
    subtitle: str = ""
    status: str = ""
    kind: str = "stage"
    metrics: list[Any] = []
    inputs: list[Any] = []
    outputs: list[Any] = []


class PipelineEdge(BaseModel):
    source: str = ""
    target: str = ""
    label: str = ""
    condition: str = ""
    active: bool = True


class PipelineDetailResponse(BaseModel):
    pipeline_key: str = ""
    updated: str = ""
    summary: dict[str, Any] = {}
    nodes: list[PipelineNode] = []
    edges: list[PipelineEdge] = []
    warnings: list[str] = []


class PipelineRegistryItem(BaseModel):
    key: str = ""
    label: str = ""
    status: str = "available"


class PipelineRegistryResponse(BaseModel):
    items: list[PipelineRegistryItem] = []
    total: int = 0

File: web/frontend/src/api/types/pipeline.ts
export interface PipelineMetric {
  label: string;
  value: string | number;
  tone?: "neutral" | "accent" | "positive" | "warning" | "negative" | string;
}

export interface PipelineNode {
  id: "inputs" | "features" | "rule_score" | "hmm_inference" | "hybrid_decision" | "stability" | "outputs" | string;
  title: string;
  subtitle: string;
  status: "ready" | "fallback" | "warning" | string;
  metrics: PipelineMetric[];
  inputs: string[];
  outputs: string[];
}

export interface PipelineEdge {
  source: string;
  target: string;
  label?: string;
}

export interface MarketRegimePipelineResponse {
  pipeline_key: "market_regime";
  updated: string;
  summary: {
    confirmed_regime: string;
    raw_regime: string;
    score: number;
    engine: string;
    detection_method: string;
    decision_reason?: string;
    confidence: number;
    entropy: number;
    adaptive_params?: Record<string, number | string>;
  };
  nodes: PipelineNode[];
  edges: PipelineEdge[];
  warnings: string[];
}
